spring-cloud-gcp-bigquery/src/main/java/org/springframework/cloud/gcp/bigquery/integration/outbound/BigQueryFileMessageHandler.java

```diff
@@ -0,0 +1,310 @@
+	 * 		the data
+	 */
+	public void setAutoDetectSchema(boolean autoDetectSchema) {
+		this.autoDetectSchema = autoDetectSchema;
+	}
+
+	/**
+	 * Sets the {@link Duration} to wait for a file to be loaded into BigQuery before timing out
+	 * when waiting synchronously.
+	 * @param timeout the {@link Duration} timeout to wait for a file to load
+	 */
+	public void setTimeout(Duration timeout) {
+		this.timeout = timeout;
+	}
+
+	/**
+	 * Sets the {@link Duration} amount of time to wait between successive polls on the status of
+	 * a BigQuery job.
+	 * @param pollInterval the {@link Duration} poll interval for BigQuery job status polling
+	 */
+	public void setJobPollInterval(Duration pollInterval) {
+		this.jobPollInterval = pollInterval;
+	}
+
+	/**
+	 * A {@code boolean} indicating if the {@link BigQueryFileMessageHandler} should synchronously
+	 * wait for each file to be successfully loaded to BigQuery.
+	 * @param sync whether {@link BigQueryFileMessageHandler} should wait synchronously for jobs to
+	 * 		complete
+	 */
+	public void setSync(boolean sync) {
+		this.sync = sync;
+	}
+
+	@Override
+	protected Object handleRequestMessage(Message<?> message) {
+		String datasetName =
+				this.datasetNameExpression.getValue(this.evaluationContext, message, String.class);
+		String tableName =
+				this.tableNameExpression.getValue(this.evaluationContext, message, String.class);
+		FormatOptions formatOptions =
+				this.formatOptionsExpression.getValue(this.evaluationContext, message, FormatOptions.class);
+
+		Assert.notNull(datasetName, "BigQuery dataset name must not be null.");
+		Assert.notNull(tableName, "BigQuery table name must not be null.");
+		Assert.notNull(formatOptions, "Data file formatOptions must not be null.");
+
+		TableId tableId = TableId.of(datasetName, tableName);
+
+		WriteChannelConfiguration writeChannelConfiguration = WriteChannelConfiguration
+				.newBuilder(tableId)
+				.setFormatOptions(formatOptions)
+				.setAutodetect(this.autoDetectSchema)
+				.setWriteDisposition(this.writeDisposition)
+				.build();
+
+		TableDataWriteChannel writer = bigQuery.writer(writeChannelConfiguration);
+
+		try (
+				InputStream inputStream = convertToInputStream(message.getPayload());
+				OutputStream sink = Channels.newOutputStream(writer)) {
+			// Write data from data input file to BigQuery
+			StreamUtils.copy(inputStream, sink);
+		}
+		catch (FileNotFoundException e) {
+			throw new MessageHandlingException(
+					message, "Failed to find file to write to BigQuery in message handler: " + this, e);
+		}
+		catch (IOException e) {
+			throw new MessageHandlingException(
+					message, "Failed to write data to BigQuery tables in message handler: " + this, e);
+		}
+
+		if (writer.getJob() == null) {
+			throw new MessageHandlingException(
+					message, "Failed to initialize the BigQuery write job in message handler: " + this);
+		}
+
+		Assert.notNull(getTaskScheduler(),
```

Comment:
Please,  move this assert into `doInit()`. This is really message-independent and would be great do not have extra logic on each requested message.

Reply:
Hmm, is there a better place to put this? I feel like if `doInit()` is called, they are likely already using Spring Integration. This warning message is more targeted to the folks who directly instantiate the object like `handler = new BigQueryFileMessageHandler(bigquery)`; and not creating it through a Spring bean.

Reply:
I moved it to the `doInit` in the latest commit - I suppose since the evaluation context initialization is there the class would be broken if the user didn't call that to begin with....

Reply:
This is not what we need to advertise. It is really wrong to use most of those components outside of Spring Context.
More over such a `MessageHandler` object is a part of Spring Integration flow and it is hard to imaging a proper use-case not wiring it into the flow and use just a standalone instance.
So, please, move it into `doInit()` and remove that `Spring Integration` message.
I would even say let's live without such an assert at all! This component should not be used outside of Spring Integration.
You indeed need to adjust your unit tests, but that's all. Just keep in mind that there is no purpose of this class outside if Spring at all

Reply:
Ok sounds good, I have removed the assert message. The tests are now autowiring the instance of the message handler instead of instantiating it with `new ...`